contracts/V1/StateChannelDiamondProxy/DisputeVerificationFacet.sol

```diff
@@ -18,26 +18,53 @@ contract DisputeVerificationFacet is StateChannelCommon {
         require(_verifyStateProof(dispute, disputeAuditingData), ErrorDisputeStateProofInvalid());
         require(_verifyExitChannelBlocks(dispute, disputeAuditingData), ErrorDisputeExitChannelBlocksInvalid());
 
-        FraudProofVerificationContext memory proofContext =
-            FraudProofVerificationContext({channelId: dispute.channelId});
-        address[] memory slashes =
-            StateChannelManagerProxy(address(this)).verifyFraudProofs(dispute.fraudProofs, proofContext);
-        slashes = StateChannelUtilLibrary.concatAddressArraysNoDuplicates(slashes, dispute.onChainSlashes);
-        address[] memory removals = _calculateRemovals(dispute);
+        // ***************** Generate output snapshot ***************
+        (SnapshotData memory outputSnapshotData, address[] memory slashes) = computeDisputeOutputSnapshotData(
+            dispute.channelId,
+            dispute.fraudProofs,
+            dispute.selfRemoval,
+            dispute.onChainSlashes,
+            dispute.disputer,
+            dispute.timeout,
+            disputeAuditingData.latestStateSnapshot,
+            disputeAuditingData.latestStateStateMachineState,
+            disputeAuditingData.genesisStateSnapshot.snapshotData.latestJoinChannelBlockHash
+        );
+
+        //verify outputStateSnapshot commitment
+        if (keccak256(abi.encode(outputSnapshotData)) != dispute.outputSnapshotDataHash) {
+            revert ErrorDisputeOutputStateSnapshotInvalid();
+        }
+
+        return slashes;
+    }
+
+    function computeDisputeOutputSnapshotData(
+        bytes32 channelId,
+        FraudProof[] memory fraudProofs,
+        bool selfRemoval,
+        address[] memory onChainSlashes,
+        address disputer,
+        Timeout memory timeout,
+        StateSnapshot memory latestStateSnapshot,
+        bytes memory latestStateMachineState,
+        bytes32 latestJoinChannelBlockHash
+    ) public returns (SnapshotData memory outputSnapshotData, address[] memory slashes) {
```

Comment:
This function just extract the pipeline from auditing in a single function so it's easy to re-use in typescript - I was lazy to refactor `DisputeStruct` into `disputeInput` and `disputeOutput` - most function run on disputeInput and you use that to calculate disputeOutput, but currently everything is jammed in - the problem is when you're creating a `disputeStruct` you don't know the `output` - you need to generate it so to reuse these functions they can't expect a disputeStruct so I just extracted what I needed in a bit ugly way

Reply:
I missed this comment :D - The link is the `generateDisputeOutputState` function - logically everything stays the same. The current `DisputeStruct` has all `input data` except `outputSnapshotDataHash` - this is the only `output` data of the `DisputeStruct`. The change that should happen is a structural change so instead of passing 10 variables 1 by 1 since we can't always pass the entire dispute, since we may not know the output data (we want to reuse this code to generate the outputData), we structure everything in a way where DisputeStruct has a field called `input` and all the input fields go there, so these functions take an argument `DisputeInput` instead of 10 separate variables